src/vorta/__main__.py

```diff
@@ -54,11 +59,44 @@ def exception_handler(type, value, tb):
         if os.fork():
             sys.exit()
 
+    # want_development is None when flag is not called
+    # it's set to the default directory when flag is called without an argument
+    # and it's set to the flag's argument if the flag has one
+    if want_development:
+        if want_development == DEFAULT_DIR_FLAG:
+            pass
+        else:
+            config.DEV_MODE_DIR = Path(want_development)
+
+        if not os.path.exists(config.DEV_MODE_DIR):
+            os.makedirs(config.DEV_MODE_DIR)
+
+        config.SETTINGS_DIR = config.DEV_MODE_DIR / 'settings'
+        if not os.path.exists(config.SETTINGS_DIR):
+            os.makedirs(config.SETTINGS_DIR)
+
+        config.LOG_DIR = config.DEV_MODE_DIR / 'logs'
+        if not os.path.exists(config.LOG_DIR):
+            os.makedirs(config.LOG_DIR)
+
+        config.CACHE_DIR = config.DEV_MODE_DIR / 'cache'
+        if not os.path.exists(config.CACHE_DIR):
+            os.makedirs(config.CACHE_DIR)
+
+        config.TEMP_DIR = config.CACHE_DIR / 'tmp'
+        if not os.path.exists(config.TEMP_DIR):
+            os.makedirs(config.TEMP_DIR)
+
+        config.PROFILE_BOOTSTRAP_FILE = config.DEV_MODE_DIR / '.vorta-init.json'
```

Comment:
Can this be moved into `vorta.config`? And can you then try to reduce redundancy that is duplicated code?

Reply:
That's what I initially planned on doing, but the result was a circular dependency I wasn't quite sure how to fix:

https://github.com/borgbase/vorta/issues/1162#issuecomment-1445214054

Reply:
This is what I came up with. However it isn't perfect since our config dirs aren't real constants any more and can't be imported with `from vorta.config import LOG_DIR` because they won't be updated when imported like that.  Instead one always has to use `from vorta import config`.

```py
SETTINGS_DIR = None
LOG_DIR = None
CACHE_DIR = None
TEMP_DIR = None
PROFILE_BOOTSTRAP_FILE = None


def default_dev_dir():
    return Path(__file__).parent.parent.parent / '.dev_config'


def init_from_platformdirs():
    dirs = platformdirs.PlatformDirs(APP_NAME, APP_AUTHOR)
    init(dirs.user_data_path, dirs.user_log_path, dirs.user_cache_path, dirs.user_cache_path / 'temp', Path.home())


def init_dev_mode(dir):
    init(dir / 'settings', dir / 'logs', dir / 'cache', dir / 'temp', dir)


def init(settings, logs, cache, temp, bootstrap):
    global SETTINGS_DIR
    global LOG_DIR
    global CACHE_DIR
    global TEMP_DIR
    global PROFILE_BOOTSTRAP_FILE
    SETTINGS_DIR = settings
    LOG_DIR = logs
    CACHE_DIR = cache
    TEMP_DIR = temp
    PROFILE_BOOTSTRAP_FILE = bootstrap / '.vorta-init.json'
    ensure_dirs()


def ensure_dirs():
    # ensure directories exist
    for dir in (SETTINGS_DIR, LOG_DIR, CACHE_DIR, TEMP_DIR):
        dir.mkdir(parents=True, exist_ok=True)
```

Reply:
Ok, so I'm gonna do this and then search through the codebase for any imports from config and modify accordingly. Sounds good?

Reply:
Ok, so I'm gonna do this and then search through the codebase for any imports from config and modify accordingly. Sounds good?

Reply:
> Ok, so I'm gonna do this and then search through the codebase for any imports from config and modify accordingly. Sounds good?

I modified all the imports except the one in `./package/vorta.spec` on line 7. That one doesn't use any of the variables that are changing. Only APP_NAME and APP_ID_DARWIN, and those are both just assigned as constants. Should I leave it like that, or should I also import the whole module for consistency's sake?